### PR TITLE
Create CONTRIBUTING.md

### LINKED ISSUE #3: Missing CONTRIBUTING file
I believe that there should be a contributing guidelines file.
Contribution guidelines are a way to quickly communicate how people should contribute, and would greatly help new contributors.
It should include things like:
# What should I know before I get started?
* Code of Conduct
* SlideShower Design Decisions
* How to use RemarkJS
# How Can I Contribute?
* Reporting Bugs
* Suggesting Enhancements
* Your First Code Contribution
* Pull Requests
* Git Commit Messages
* JavaScript Styleguide
* PHP Anti-Styleguide
* Documentation Styleguide
# Additional Notes
* Which Traditional Welsh Songs Should Be Sung Whilst Developing
* List Of Forbidden Songs
* Issue and Pull Request Labels

### PR DESCRIPTION
**Pull request by danprince**
Ticking off the main points that new contributors need to be mindful of for this project. Closes #3

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+* Roll a red tuffet into a printer.
+* Prevent the cleaners from waking up any sleeping members of the core development team.
+* Announce the opening of Pontio.
+* Summon Richard Stallman through a non-free overhead projector
+
+## Reporting Bugs
+Bugs should be reported through the internal JIRA instance. 
+
+## Suggesting Enhancements
+Enhancements should be sent by Royal Mail First Class Recorded Delivery to: P. B. Butcher, 2 Patterdale, Cumbria, CA32 9FS.
```
Comment (eeue56) [inline CONTRIBUTING.md:15]: @PButcher can you confirm that the house martin has flown the nest?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+# What should I know before I get started?
+* PHP
+* The fastest route between main arts and late stop.
+
+# How Can I Contribute?
+* Roll a red tuffet into a printer.
+* Prevent the cleaners from waking up any sleeping members of the core development team.
+* Announce the opening of Pontio.
+* Summon Richard Stallman through a non-free overhead projector
```
Comment (eeue56) [inline CONTRIBUTING.md:9]: I believe the correct terminology is to "force lift".

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+## Reporting Bugs
+Bugs should be reported through the internal JIRA instance.
```
Comment (eeue56) [inline CONTRIBUTING.md:12]: I believe the correct first step is to generate a PDF from Latex in order to report bugs. Then they will be transferred by less important people into JIRA. Also, ensure to pronounce JIRA as `gye-ra`

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+Enhancements should be sent by Royal Mail First Class Recorded Delivery to: P. B. Butcher, 2 Patterdale, Cumbria, CA32 9FS.
+
+## Your First Code Contribution
+TODO
+
+## Pull Requests
+TODO
+
+## Git Commit Messages
+A descriptive commit message should also be enjoyable and insightful to read. For this project, we're going with haiku commits.
```
Comment (eeue56) [inline CONTRIBUTING.md:24]: I prefer thoughtprovoking to insightful commit messsages

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+## JavaScript Styleguide
+In order to stick to the good parts of JavaScript, we'll be using an eslint config setup to ignore all language constructs and standard library, other than ES6 Symbols, `with`, `do..for`, `let` and `Math.LOG2E`.
+
+## PHP Anti-Styleguide
+Going forward we'll be using the [Yay macro library](https://github.com/marcioAlmada/yay) so that we can transpile Elm into macroized PHP for the backend.
```
Comment (eeue56) [inline CONTRIBUTING.md:30]: This is the only reasonable stack for a project of this quality

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+## Documentation Styleguide
+Documentation should be submitted as rasterized PNGs using Comic Sans Neue as the font. For a guide on getting started with our documentation editor, check out [this video](https://www.youtube.com/watch?v=hlnodEYZ71w).
```
Comment (eeue56) [inline CONTRIBUTING.md:33]: Please ensure that all web development assignments are submitted as screenshots drawn in paint only.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+# Additional Notes
+## Which Traditional Welsh Songs Should Be Sung Whilst Developing
+* Fuck the Pain Away
+* Tan Yn Llyn
+* Sosban Fawr
+
+## List Of Forbidden Songs
```
Comment (eeue56) [inline CONTRIBUTING.md:41]: I believe the correct terminology is "vox populi"

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+TODO
+
+## Pull Requests
+TODO
+
+## Git Commit Messages
+A descriptive commit message should also be enjoyable and insightful to read. For this project, we're going with haiku commits.
+
+## JavaScript Styleguide
+In order to stick to the good parts of JavaScript, we'll be using an eslint config setup to ignore all language constructs and standard library, other than ES6 Symbols, `with`, `do..for`, `let` and `Math.LOG2E`.
```
Comment (subsidel) [inline CONTRIBUTING.md:27]: Could you add a line about the javascript libraries? Something along the lines of that we will be transitioning away from using JavaCoding to [Scala.js](https://www.scala-js.org/) with [JSXQuery](https://www.npmjs.com/package/jsxquery)?

**Review by subsidel — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (PButcher) [reply CONTRIBUTING.md:15]: How come I only just saw this... :'D

**Pull request closed without merging**